### ISSUE TITLE
Allow preLaunchTask to just check running task

### ISSUE THREAD

**Issue by octref** [Microsoft/vscode#50360]
One of my pain points of dealing with tasks is many `launch.json` config specify a `preLaunchTask` that is slow. However, these tasks can be fast if they are run using watch, because of incremental compilation of TypeScript.

We don't have any `preLaunchTask` in vscode repo, probably because starting compilation anew is taking 1 minute. Everyone runs their build task on terminal. So when making changes compilation finish fast and developer can instantly launch debug.

It would be great if I can start a task `build` running `tsc --watch` (or gulp or any other watch task) to incrementally build my project, and still have `preLacunTask` as `build`, so the behavior is:

- When the `build` task is not running, start `build` task and launch debug session.
- When the `build` task is already running, do nothing and launch debug session immediately.

**Comment by dbaeumer** [comment]
@isidorn I think this is supported with the only difference that when such a task is launch the first time debug waits until the first compile path is finished. But I think we could easily change this.

**Comment by isidorn** [comment]
This is working just fine for me.
@octref To clarify your two points
* When the build task is not running, start build task, wait for it to compile and launch debug session
* When the build task is already running, do nothing and launch debug session immediately.

If you are not seeing this please provide a repository for which you do not see this behavior. Example which I tried and works fine is a VSCode typescript extension

**Comment by octref** [comment]
@isidorn 

See https://github.com/Microsoft/vscode/tree/2272dab6da9ed08b82a95d136c4fe41507848e04/extensions/css-language-features/.vscode

My current situation:

1. When I use a watch task, such as our `gulp watch-extension-<ext>`, it won't exit. No matter if I'm using `$gulp-tsc` or `$tsc-watch` matcher, debug session won't launch:

  ![image](https://user-images.githubusercontent.com/4033249/40732004-2c83ea4e-63e7-11e8-8908-a5694de55c81.png)

2. If I do run the task by myself, and then launch the debug session, indeed it'll skip the `preLaunchTask`.

What I want to achieve:

1. Have a watch task as `preLaunchTask`
2. When the task is not being run and I start a debug session
3. The watch task should start
4. When the watch task finishes first run, launch debug session
5. The watch task is then "detached" and kept running
6. As long as the task hasn't been terminated, I can launch debug session while skipping the `preLaunchTask`


@dbaeumer I also found you are commenting out the `preLaunchTask` as I did. Are you having the same problem?

https://github.com/Microsoft/vscode-eslint/blob/master/.vscode/launch.json

```json
// A launch configuration that compiles the extension and then opens it inside a new window
{
	"version": "0.2.0",
	"configurations": [
		{
			"type": "extensionHost",
			"request": "launch",
			"name": "Launch Extension",
			"runtimeExecutable": "${execPath}",
			"args": ["--extensionDevelopmentPath=${workspaceRoot}" ],
			"stopOnEntry": false,
			"sourceMaps": true,
			"outFiles": ["${workspaceRoot}/client/out/**/*.js"]
			// "preLaunchTask": "watch"
		},
		{
			"type": "node",
			"request": "attach",
			"name": "Attach to Server",
			"address": "localhost",
			"protocol": "inspector",
			"port": 6010,
			"sourceMaps": true,
			"outFiles": ["${workspaceRoot}/server/**/*.js"]
		}
	]
}
```
